packages/webpack/test/e2e-scuttle.spec.js

```diff
@@ -0,0 +1,87 @@
+const test = require('ava')
+const path = require('node:path')
+// eslint-disable-next-line ava/no-import-test-files
+const { scaffold, runScriptWithSES } = require('./scaffold.js')
+const {makeConfig} = require('./fixtures/main/webpack.config.js')
+
+const err = (intrinsic) =>
+  'LavaMoat - property "' +
+  intrinsic +
+  '" of globalThis is inaccessible under ' +
+  'scuttling mode. To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+
+async function scuttle(t, scuttleGlobalThis, globals) {
+  const webpackConfigDefault = makeConfig({
+    scuttleGlobalThis,
+    generatePolicy: true,
+    diagnosticsVerbosity: 1,
+    policyLocation: path.resolve(__dirname, 'fixtures/main/policy-scuttling'),
+  })
+  const webpackConfig = {
+    ...webpackConfigDefault,
+    entry: {
+      app: './simple.js',
+    },
+  }
+  await t.notThrowsAsync(async () => {
+    t.context.build = await scaffold(webpackConfig)
+    t.context.bundle = t.context.build.snapshot['/dist/app.js']
+    t.context.globalThis = runScriptWithSES(t.context.bundle, globals).context
+  }, 'Expected the build to succeed')
+}
+
+test(`webpack/scuttled - dist shape`, async (t) => {
+  await scuttle(t, true)
+  t.snapshot(Object.keys(t.context.build.snapshot))
+})
+
+test(`webpack/scuttled - hosting globalThis's environment is not scuttled`, async (t) => {
+  await scuttle(t)
+  try {
+    const global = t.context.globalThis
+    Object.getOwnPropertyNames(global).forEach(name => global[name])
+  } catch (e) {
+    t.fail(`Unexpected error in scenario: ${e.message}`)
+  }
```

Comment:
(suggestion) `try` / `catch` / `t.fail()` this is better expressed `t.notThrows(fn)`.  I'd normally just omit the assertion altogether, but I think AVA treats a test case w/o an assertion as an error.

Reply:
like this? 4e41f842e

Reply:
yes exactly like that 😄